peacebot/core/plugins/Moderation/mod.py

```diff
@@ -0,0 +1,259 @@
+            timestamp=datetime.now().astimezone(),
+        )
+        .set_author(
+            name=f"{ctx.author}(ID {ctx.author.id})", icon=ctx.author.avatar_url
+        )
+        .set_thumbnail(member.avatar_url)
+    )
+    await mod_logs.send(embed=embed)
+    response = await ctx.respond(embed=embed)
+    message_link = (await response.message()).make_link(ctx.guild_id)
+    await register_cases(
+        guild_id=ctx.guild_id,
+        moderator=ctx.author.id,
+        target=member.id,
+        reason=ctx.options.reason,
+        message_link=message_link,
+        channel_id=ctx.channel_id,
+        type="Kick",
+    )
+
+
+@mod_plugin.command
+@lightbulb.option(
+    "days",
+    "Specify number of days of messages to delete(Min - 0, Max- 7)",
+    required=False,
+    type=int,
+)
+@lightbulb.option("reason", "The reason to ban the member")
+@lightbulb.option("member", "The member to ban", type=hikari.Member)
+@lightbulb.command("ban", "Ban the member from the server")
+@lightbulb.implements(lightbulb.SlashCommand, lightbulb.PrefixCommand)
+@mod_role_check
+async def ban_command(ctx: lightbulb.Context) -> None:
+    member: hikari.InteractionMember = ctx.options.member
+    higher_role_check(ctx.member, member)
+    mod_logs = await mod_logs_check(ctx)
+    if ctx.options.days > 7:
+        raise ModerationError("Cannot delete messages more than 7 days old!")
+    await member.ban(
+        delete_message_days=ctx.options.days or 0, reason=ctx.options.reason
+    )
+    embed = (
+        hikari.Embed(
+            description=f"🔨 Banned -> {member}\n**Reason:** {ctx.options.reason}",
+            color=EmbedColors.ERROR,
+            timestamp=datetime.now().astimezone(),
+        )
+        .set_author(
+            name=f"{ctx.author}(ID {ctx.author.id})", icon=ctx.author.avatar_url
+        )
+        .set_thumbnail(member.avatar_url)
+    )
+    await mod_logs.send(embed=embed)
+    response = await ctx.respond(embed=embed)
+    message_link = (await response.message()).make_link(ctx.guild_id)
+    await register_cases(
+        guild_id=ctx.guild_id,
+        moderator=ctx.author.id,
+        target=member.id,
+        reason=ctx.options.reason,
+        message_link=message_link,
+        channel_id=ctx.channel_id,
+        type="Ban",
+    )
+
+
+@mod_plugin.command
+@lightbulb.command("case", "Group for case commands")
+@lightbulb.implements(lightbulb.SlashCommandGroup, lightbulb.PrefixCommandGroup)
+async def case_command(ctx: lightbulb.Context) -> None:
+    await ctx.bot.help_command.send_group_help(ctx, ctx.command)
+
+
+@case_command.child
+@lightbulb.option("case_id", "Case ID to look for")
+@lightbulb.command("search", "Get a specific case")
+@lightbulb.implements(lightbulb.SlashSubCommand, lightbulb.PrefixSubCommand)
+async def case_search_command(ctx: lightbulb.Context) -> None:
+    model = await ModLogs.get_or_none(guild_id=ctx.guild_id, id=ctx.options.case_id)
+    if model is None:
```

Comment:
It'd be more pythonic to use
```py
if not model:
    ...
```

Reply:
Well not being more pythonic is more feasible for this use case, as we are raising an error if the model is `None` and then performing other operations.

Reply:
Can you explain? @thenishantsapkota

Reply:
Well, if the `ModLogs` model has no data with the specific ID in it, we are raising a `ModerationError`
That's why I have chose to do `if model is None -> Raise error` and perform the rest of the operations!